### PR TITLE
move Spline to single precision 

### PR DESCRIPTION
**Pull request by davidlange6**
to save 50MB in reco (we should also understand why we have 50MB of splines)
Thanks to Danilo for noticing this.

[I didn't check the physics changes - please do before signing]

### CODE REVIEW

File: PhysicsTools/MVAComputer/src/Spline.cc [hunk 1]
```diff
@@ -29,20 +29,20 @@ double Spline::Segment::eval(double x) const
...
 
-double Spline::Segment::deriv(double x) const
+float Spline::Segment::deriv(float x) const
 {
-	double tmp;
-	double d = 0.0;
+	float tmp;
+	float d = 0.0;
 	d += coeffs[1];			tmp = x;
 	d += coeffs[2] * tmp * 2.0;	tmp *= x;
```
Comment (VinInn) [inline PhysicsTools/MVAComputer/src/Spline.cc]: better to change all literals to 2.0f, 3.0f etc...

**Comment by VinInn** [comment]
still 50MB of floats is 12.5M values: to do what?

File: PhysicsTools/MVAComputer/src/Spline.cc [hunk 2]
```diff
@@ -18,10 +18,10 @@
...
 namespace PhysicsTools {
 
-double Spline::Segment::eval(double x) const
+float Spline::Segment::eval(float x) const
 {
-	double tmp;
-	double y = 0.0;
+	float tmp;
+	float y = 0.0;
 	y += coeffs[0];		tmp = x;
```
Comment (VinInn) [inline PhysicsTools/MVAComputer/src/Spline.cc]: Horner, the Unknown...

**Comment by davidlange6** [comment]
@cmsbuild, please test
(the comparison for this PR seems broken)

**Comment by davidlange6** [comment]
Will merge - Only diffs are off by one bin issues in 

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_7_4_X_2015-02-11-0200+7632/6011/validateJR/all_OldVSNew_TTbarFSwf5p1/all_OldVSNew_TTbarFSwf5p1c_recoJetedmRefToBaseProdTofloatsAssociationVector_combinedInclusiveSecondaryVertexV2BJetTags__HLT_obj_data_83.png

**Pull request merged**